src/error.rs

```diff
@@ -1,27 +1,37 @@
 #![allow(missing_docs)]
 #[cfg(all(unix, not(target_os = "macos")))]
 use dbus;
+use std::num;
 
-#[cfg(target_os = "macos")]
-use mac_notification_sys;
+pub type Result<T> = ::std::result::Result<T, Error>;
+pub use failure::Error;
 
-use std::num;
+#[derive(Fail, Debug)]
+pub enum ErrorKind {
+    #[fail(display="{}", _0)]
+    #[cfg(all(unix, not(target_os = "macos")))]
+    Dbus(#[cause] dbus::Error),
 
-error_chain!{
-    types {
-        Error, ErrorKind, ResultExt, Result;
-    }
+    #[fail(display="Parsing Error")]
+    Parse(#[cause] num::ParseIntError),
+
+    #[fail(display="The running server supplied an unknown version: {}", _0 )]
+    SpecVersion(String),
+
+    #[fail(display="ParseError: {}", error)]
+    ParseError{ #[cause] error: ::std::num::ParseIntError },
```

Comment:
Hi @hoodie, I have been reading up on the failure crate, and the changes here look fine.

There is only one thing I do not yet understand. Why is there `Parse` and `ParseError`, when both store a `::std::num::ParseIntError`? To me, it looks like the later one would be sufficient and should be returned by `From<num::ParseIntError>`.